src/extension.ts

```diff
@@ -147,35 +146,53 @@ class RunLineInTerminal {
     }
 }
 
+const elegantSpinner = require('elegant-spinner');
+
 class RunLineInEditor {
 
     private resultDocument: TextDocument | undefined;
     private parsedResult: object | undefined;
     private queryEnabled = false;
     private query: string | undefined;
     private disposables: Disposable[] = [];
+    private readonly settings: AzureCliToolsSettings = AzureCliToolsSettings.Instance;
+    private runStatusBarItem: StatusBarItem;
+    private interval!: NodeJS.Timer;
+    private spinner = elegantSpinner();
 
     constructor(private status: StatusBarInfo) {
         this.disposables.push(commands.registerTextEditorCommand('ms-azurecli.toggleLiveQuery', editor => this.toggleQuery(editor)));
         this.disposables.push(commands.registerTextEditorCommand('ms-azurecli.runLineInEditor', editor => this.run(editor)));
         this.disposables.push(workspace.onDidCloseTextDocument(document => this.close(document)));
         this.disposables.push(workspace.onDidChangeTextDocument(event => this.change(event)));
+
+        this.runStatusBarItem = window.createStatusBarItem(StatusBarAlignment.Left);
```

Comment:
Could you do this in a separate PR? And maybe hide the status bar item 10 seconds after the command finished, so it doesn't use up space in the status bar when it is no longer needed.

Reply:
This has been separated into https://github.com/microsoft/vscode-azurecli/pull/56